amethyst_renderer/src/pipe/effect/pso.rs

```diff
@@ -41,56 +41,99 @@ impl<'d> PipelineInit for Init<'d> {
     fn link_to<'r>(&self, desc: &mut Descriptor, info: &'r ProgramInfo) -> InitResult<'r, Meta> {
         let mut meta = Meta::default();
 
-        for (info, cbuf) in info.constant_buffers.iter().zip(&self.const_bufs) {
+        for cbuf in self.const_bufs.iter() {
             let mut meta_cbuf = <RawConstantBuffer as DataLink<'d>>::new();
-            if let Some(res) = meta_cbuf.link_constant_buffer(info, cbuf) {
-                let d = res.map_err(|e| InitError::ConstantBuffer(info.name.as_str(), Some(e)))?;
-                meta.const_bufs.push(meta_cbuf);
-                desc.constant_buffers[info.slot as usize] = Some(d);
+            for info in info.constant_buffers.iter() {
+                // println!("Link constant {:?}/{:?}", info.name, cbuf);
+                if let Some(res) = meta_cbuf.link_constant_buffer(info, cbuf) {
+                    // println!("Linked {:?}", res);
+                    let d = res.map_err(|e| InitError::ConstantBuffer(info.name.as_str(), Some(e)))?;
+                    desc.constant_buffers[info.slot as usize] = Some(d);
+                    break;
+                }
             }
+            meta.const_bufs.push(meta_cbuf);
```

Comment:
I made this change intentionally.
If meta wouldn't pushed then all data after it will be at wrong indexes at binding phase